component/common/net/server.go

```diff
@@ -1,97 +1,353 @@
 package net
```

Comment:
This is a modified copy of weaveworks' common server. I've removed the GRPC support and made it work with our merged configuration struct.